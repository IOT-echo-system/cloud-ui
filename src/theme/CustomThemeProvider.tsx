import type {PropsWithChildren} from 'react'
import React from 'react'
import {ThemeProvider} from '@mui/system'
import {useSelector} from '@/hooks'
import dark from './dark'
import light from './light'

const CustomThemeProvider: React.FC<PropsWithChildren> = ({children}) => {
  const site = useSelector(state => state.site)
  return <ThemeProvider theme={site.theme === 'dark' ? dark : light}>{children}</ThemeProvider>
}

export default CustomThemeProvider
