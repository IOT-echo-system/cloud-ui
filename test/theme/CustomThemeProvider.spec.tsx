import React from 'react'
import {render} from '@testing-library/react'
import CustomThemeProvider from '@/theme/CustomThemeProvider'

jest.mock('../../src/hooks/useStore')
describe('Custom Theme Provider Test', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })
  afterEach(jest.resetAllMocks)

  it('should match with the snapshot for light theme', () => {
    // const site = siteBuilder({theme: 'light'})
    // const globalState = globalStateBuilder({site})
    // const useSelectorSpy = jest.spyOn(GlobalStore, 'useSelector').mockReturnValue(globalState)

    const {container} = render(<CustomThemeProvider />)

    expect(container).toMatchSnapshot()
    // expect(GlobalStore.useSelector).toHaveBeenCalledTimes(1)
    // expect(GlobalStore.useSelector).toHaveBeenCalledWith(expect.any(Function))
    //
    // const useSelectorCallback = useSelectorSpy.mock.calls[0][0]
    // expect(useSelectorCallback(globalState)).toStrictEqual(site)
  })

  // it('should match with the snapshot for dark theme', () => {
  //   const site = siteBuilder({theme: 'dark'})
  //   const globalState = globalStateBuilder({site})
  //   jest.spyOn(GlobalStore, 'useSelector').mockReturnValue(globalState)
  //
  //   const {container} = render(<CustomThemeProvider />)
  //
  //   expect(container).toMatchSnapshot()
  //   expect(GlobalStore.useSelector).toHaveBeenCalledTimes(1)
  //   expect(GlobalStore.useSelector).toHaveBeenCalledWith(expect.any(Function))
  // })
})
