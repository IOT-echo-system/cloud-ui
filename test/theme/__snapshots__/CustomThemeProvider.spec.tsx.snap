// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Theme Provider Test should match with the snapshot for light theme 1`] = `<div />`;
